libandroid-navigation/src/test/java/com/mapbox/navigation/DefaultDirectionsSessionTest.kt

```diff
@@ -18,29 +19,34 @@ class DefaultDirectionsSessionTest {
 
     private val router: Router = mockk(relaxUnitFun = true)
     private val origin: Point = mockk(relaxUnitFun = true)
+    private val destination: Point = mockk(relaxUnitFun = true)
     private val waypoints: List<Point> = mockk(relaxUnitFun = true)
     private val observer: DirectionsSession.RouteObserver = mockk(relaxUnitFun = true)
-    private lateinit var routeCallback: ((route: Route) -> Unit)
+    private lateinit var routeListener: Router.RouteListener
```

Comment:
We're missing a couple of test cases where we request the route, capture the callback and invoke it (either onSuccess or onError). We should verify what happens in those scenarios.

Reply:
good catch, test was added